packages/medusa/src/types/routing.ts

```diff
@@ -61,6 +62,16 @@ export interface MedusaRequest<Body = unknown>
    * A generic context object that can be used across the request lifecycle
    */
   context?: Record<string, any>
+  /**
+   * A custom validator for the request body that will be merged with the original
+   * validator of the route
+   */
+  bodyValidator?: ZodObject<any, any>
```

Comment:
NIT: The naming maybe can be better, as like this is sounds like it's the only validator, unless you read the description. I am raising more to see what you think, as I don't have great ideas, `extendedBodySchema` and `extendedQuerySchema` come to mind.

Reply:
Yeah. I agree, the `bodyValidator` seems like the final validator to be applied and not merged.

We intentionally decided to not use the keyword `schema` because it might mean a query schema, a db schema and so on.

So maybe, we can rename the property to `validatorsToMerge.body` and `validatorsToMerge.queryParams`. Wdyt?

Reply:
Something like this 

```ts
req.validatorsToMerge = {
  body: zod.object(),
  queryParams: zod.object(),
}
```

Reply:
I was fine with the current name as it can evolve later, eventually I was thinking of something like:
```ts
type ExtendedValidator = ZodObject<any, any> | {
  schema: ZodObject<any, any>,
  strategy?: 'merge' | 'override'
}

req.extendedValidator = {}
```

but we can argue that we don't need that right now and can leave that for the future

Reply:
very-nit and totally non-blocking: How about? Would bring their name very close to what they are 😄 
```
validatorExtensions.body
validatorExtensions.params
```

Reply:
Went with `extendedValidators` suggestion